spec/request_interceptor_spec.rb

```diff
@@ -125,4 +125,70 @@
       expect(response.body).to match /Stack Overflow/im
     end
   end
+
+  context 'with a custom class as application template' do
+    let(:template) do
+      Class.new(Sinatra::Application) do
+        def answer
+          42
+        end
+      end
+    end
+
+    around do |spec|
+      default_template = RequestInterceptor.template
+      RequestInterceptor.template = template
+      spec.run
```

Comment:
I'm not too familiar with how rspec works, but if the spec fails / raises, will the next line still be executed?